pkg/auth/providers/saml/saml_provider.go

```diff
@@ -243,21 +336,20 @@ func (s *Provider) SearchPrincipals(searchKey, principalType string, token v3.To
 	}
 
 	principals = append(principals, p)
-
 	return principals, nil
 }
 
 func (s *Provider) GetPrincipal(principalID string, token v3.Token) (v3.Principal, error) {
-	parts := strings.SplitN(principalID, ":", 2)
-	if len(parts) != 2 {
+	externalID, principalType := splitPrincipalID(principalID)
+	if externalID == "" && principalType == "" {
 		return v3.Principal{}, fmt.Errorf("SAML: invalid id %v", principalID)
 	}
-	principalType := parts[0]
-	externalID := strings.TrimPrefix(parts[1], "//")
-
 	if principalType != s.userType && principalType != s.groupType {
 		return v3.Principal{}, fmt.Errorf("SAML: Invalid principal type")
 	}
+	if principalType == s.groupType && s.hasLdapGroupSearch() {
```

Comment:
GetPrincipal should also be called for users I think?

Reply:
we are only trying to shim in the group functionality to a SAML provider and I'm trying to leave the user and login code mostly alone.